src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,446 @@
+            if (attributeType == PlatformAttributeType.ObsoletedInOSPlatformAttribute)
+                return ObsoleteRule;
+            return RemovedRule;
+        }
+
+        private static AttributeData[] FindAllPlatformAttributesApplied(ImmutableArray<AttributeData> immediateAttributes, INamedTypeSymbol parent)
+        {
+            using var builder = ArrayBuilder<AttributeData>.GetInstance();
+            foreach (AttributeData attribute in immediateAttributes)
+            {
+                if (s_osPlatformAttributes.Contains(attribute.AttributeClass.Name))
+                {
+                    builder.Add(attribute);
+                }
+            }
+
+            while (parent != null)
+            {
+                var current = parent.GetAttributes();
+                foreach (var attribute in current)
+                {
+                    if (s_osPlatformAttributes.Contains(attribute.AttributeClass.Name) &&
+                        TargetPlatformAttribute != attribute.AttributeClass.Name)
+                    {
+                        builder.Add(attribute);
+                    }
+                }
+                parent = parent.BaseType;
+            }
+            return builder.ToArray();
+        }
+
+        private static bool IsSuppressedByAttribute(PlatformAttributeInfo diagnosingAttribute, ISymbol containingSymbol)
+        {
+            while (containingSymbol != null)
+            {
+                var attributes = containingSymbol.GetAttributes();
+                if (attributes != null)
+                {
+                    foreach (AttributeData attribute in attributes)
+                    {
+                        if (diagnosingAttribute.AttributeType.ToString() == attribute.AttributeClass.Name)
+                        {
+                            if (PlatformAttributeInfo.TryParsePlatformAttributeInfo(attribute, out PlatformAttributeInfo parsedAttribute))
+                            {
+                                if (IsOSPlatformsEqual(diagnosingAttribute.PlatformName, parsedAttribute.PlatformName) &&
+                                    AttributeVersionsMatch(diagnosingAttribute, parsedAttribute))
+                                {
+                                    return true;
+                                }
+                            }
+                        }
+                    }
+                }
+                containingSymbol = containingSymbol.ContainingSymbol;
+            }
+            return false;
+        }
+
+        private static bool AttributeVersionsMatch(PlatformAttributeInfo diagnosingAttribute, PlatformAttributeInfo osAttribute)
+        {
+            return AttributeVersionsMatch(diagnosingAttribute.AttributeType, diagnosingAttribute.Version, osAttribute.Version);
+        }
+
+        private static bool AttributeVersionsMatch(PlatformAttributeType attributeType, Version diagnosingVersion, Version suppressingVersion)
```

Comment:
Name of the method is misleading - we are not checking for a match, but rather feasibility. Say something like `bool CheckVersionRequirement(PlatformAttributeType attributeType, Version minimumRequiredVersion, Version actualVersion)`

Reply:
With new requirement, we gonna check version range so this method gonna change